src/cpu/aarch64/brgemm/brgemm.cpp

```diff
@@ -0,0 +1,566 @@
+    bool ldx_check = (brg->is_row_major()) ? (LDA < K)
+                                           : (LDA < M || LDB < K || LDC < M);
+    if (ldx_check) return status::invalid_arguments;
+
+    if (utils::everyone_is(
+                false, brg->is_int8, brg->is_bf16, brg->is_f32, brg->is_f16))
+        return status::unimplemented;
+
+    CHECK(brgemm_blocking(brg));
+
+    return status::success;
+}
+
+status_t brdgmm_desc_init(brgemm_t *brg, cpu_isa_t isa,
+        brgemm_batch_kind_t type, impl::data_type_t dt_a,
+        impl::data_type_t dt_b, bool transA, brgemm_layout_t layout,
+        float alpha, float beta, dim_t LDA, dim_t LDC, dim_t M, dim_t N,
+        const brgemm_strides_t *strides) {
+
+    if (brg == nullptr) return status::invalid_arguments;
+    if (transA || layout != brgemm_row_major || alpha != 1.0f || beta != 0.f)
+        return status::unimplemented;
+
+    brgemm_utils::init_brdgmm_conf(brg, isa, type, dt_a, dt_b, layout, alpha,
+            beta, LDA, LDC, M, N, strides);
+
+    const bool ldx_check = (LDA < N || LDC < N);
+    if (ldx_check) return status::invalid_arguments;
+
+    if (utils::everyone_is(
+                false, brg->is_int8, brg->is_bf16, brg->is_f32, brg->is_f16))
+        return status::unimplemented;
+
+    CHECK(brdgmm_blocking(brg));
+
+    return status::success;
+}
+
+status_t brgemm_desc_set_postops(brgemm_t *brg, const primitive_attr_t *attr,
+        const memory_desc_t *dst_md, int LDD, impl::data_type_t dt_bias) {
+    if (!brg || !dst_md) return status::invalid_arguments;
+
+    brg->attr = attr;
+    brg->dst_md = dst_md;
+
+    brg->with_bias = (dt_bias == data_type::undef) ? false : true;
+    brg->dt_bias = dt_bias;
+    brg->typesize_bias = (dt_bias == data_type::undef)
+            ? 0
+            : types::data_type_size(brg->dt_bias);
+
+    brg->LDD = LDD;
+    const auto dt_d = dst_md->data_type;
+
+    if ((brg->dt_a == data_type::u8 && brg->dt_b == data_type::s8)
+            && (!one_of(dt_d, data_type::u8, data_type::s8, data_type::s32,
+                    data_type::f32))
+            && (!one_of(dt_bias, data_type::undef, data_type::u8, data_type::s8,
+                    data_type::s32, data_type::f32, data_type::bf16)))
+        return status::unimplemented;
+    if ((brg->dt_a == data_type::bf16 && brg->dt_b == data_type::bf16)
+            && (!one_of(dt_d, data_type::bf16, data_type::f32))
+            && (!one_of(dt_bias, data_type::undef, data_type::bf16,
+                    data_type::f32)))
+        return status::unimplemented;
+    if ((brg->dt_a == data_type::f32 && brg->dt_b == data_type::f32)
+            && (!one_of(dt_d, data_type::f32))
+            && (!one_of(dt_bias, data_type::undef, data_type::f32)))
+        return status::unimplemented;
+
+    brg->dt_d = dt_d;
+    brg->typesize_D = types::data_type_size(brg->dt_d);
+
+    if (brg->is_int8 || (brg->dt_d == bf16)) return status::unimplemented;
```

Comment:
@vineelabhinav 
(nit): Do you plan to support int8/bf16 long term? Currently it is not supported but there is a lot of code relevant only for those data types.

Reply:
@igorsafo Yes we will implement in future.